tests/messages_tests/tests.py

```diff
@@ -63,6 +64,19 @@ class TestLevelTags(SimpleTestCase):
     def test_override_settings_level_tags(self):
         self.assertEqual(base.LEVEL_TAGS, self.message_tags)
 
+    def test_lazy(self):
+        # Don't use @override_settings to avoid calling the setting_changed
+        # signal.
+        old_message_tags = getattr(settings, "MESSAGE_TAGS", None)
+        settings.MESSAGE_TAGS = {constants.ERROR: "bad"}
+        try:
+            self.assertEqual(base.LEVEL_TAGS[constants.ERROR], "bad")
+        finally:
+            if old_message_tags is None:
+                del settings.MESSAGE_TAGS
+            else:
+                settings.MESSAGE_TAGS = old_message_tags
```

Comment:
I'm afraid that it's not possible to test this regression as other tests override `MESSAGE_TAGS`.

Reply:
This is a proper regression test when run in isolation.

Reply:
OK, "maybe" I found the solution :crossed_fingers:

Reply:
Thanks for working on this! what do you mean with "when run in isolation"? if I comment out the fix, the test does fail when running the whole set (`./runtests.py messages_tests`), but when run this single test (`./runtests.py messages_tests.tests.TestLevelTags.test_lazy`) it does not fail.

Reply:
I did some investigation for this test. My conclusions are as follows:

When the test `messages_tests.tests.TestLevelTags.test_lazy` is run in isolation, the `settings_changed` signal is never sent so the `update_level_tags` signal handler is never executed in the test (I double checked this by adding prints inside the `update_level_tags` method).

So actually calling `override_settings` and really triggering the `setting_changed` signal is what I think would make the test more robust, I have confirmed this test fails/pass accordingly both when run as a single unit or when run with the rest of the suite if the `override_settings` is called *and* we ensure that `constants.ERROR` has a different value than the one tested:

```python
    @override_settings(MESSAGE_TAGS=message_tags)
    def test_lazy(self):
        # The `update_level_tags` handler has been called at least once before
        # running this code when using `@override_settings`.
        settings.MESSAGE_TAGS = {constants.ERROR: "very-bad"}
        self.assertEqual(base.LEVEL_TAGS[constants.ERROR], "very-bad")
```

The above test fails with (both when ran alone or with the rest of the suite):

```
======================================================================
FAIL: test_lazy (messages_tests.tests.TestLevelTags.test_lazy)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nessita/fellowship/django/django/test/utils.py", line 443, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/nessita/fellowship/django/tests/messages_tests/tests.py", line 72, in test_lazy
    self.assertEqual(base.LEVEL_TAGS[constants.ERROR], "very-bad")
AssertionError: 'bad' != 'very-bad'
- bad
+ very-bad
```

Reply:
> Thanks for working on this! what do you mean with "when run in isolation"? 

I marked my previous comment as resolved, so IMO it should work now.

> if I comment out the fix, the test does fail when running the whole set (`./runtests.py messages_tests`), but when run this single test (`./runtests.py messages_tests.tests.TestLevelTags.test_lazy`) it does not fail.

That's weird. For me, a regression test crashes in all cases on the current `main`:
```
./runtests.py
./runtests.py messages_tests
./runtests.py messages_tests.tests.
./runtests.py messages_tests.tests.TestLevelTags
./runtests.py messages_tests.tests.TestLevelTags.test_lazy
```

Reply:
Thanks for pointing this out, you are correct that your test fails when the changes to `base` are reverted.

Though we may still need a regression test for the changes to `apps`, perhaps we can use my test for it? After our conversation I realized we may need the two tests, since each test covers different parts of the fix.

What do you think?

Reply:
Added.